app/src/main/java/com/example/wordsapp/LetterListFragment.kt

```diff
@@ -0,0 +1,140 @@
+        val view = binding.root
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        recyclerView = binding.recyclerView
+        // Sets the LayoutManager of the recyclerview
+        // On the first run of the app, it will be LinearLayoutManager
+        chooseLayout()
+    }
+
+    /**
+     * Sets the LayoutManager for the [RecyclerView] based on the desired orientation of the list.
+     *
+     * Notice that because the enclosing class has changed from an Activity to a Fragment,
+     * the signature of the LayoutManagers has to slightly change.
+     */
+    private fun chooseLayout() {
+        if (isLinearLayoutManager) {
+            recyclerView.layoutManager = LinearLayoutManager(context)
+        } else {
+            recyclerView.layoutManager = GridLayoutManager(context, 4)
+        }
+        recyclerView.adapter = LetterAdapter()
+    }
+
+    private fun setIcon(menuItem: MenuItem?) {
+        if (menuItem == null)
+            return
+
+        // Set the drawable for the menu icon based on which LayoutManager is currently in use
+
+        // An if-clause can be used on the right side of an assignment if all paths return a value.
+        // The following code is equivalent to
+        // val context = this.requireContext()
+        // if (isLinearLayoutManager)
+        //     menu.icon = ContextCompat.getDrawable(context, R.drawable.ic_grid_layout)
+        // else menu.icon = ContextCompat.getDrawable(context, R.drawable.ic_linear_layout)
+        menuItem.icon =
+            if (isLinearLayoutManager)
+                ContextCompat.getDrawable(this.requireContext(), R.drawable.ic_grid_layout)
+            else ContextCompat.getDrawable(this.requireContext(), R.drawable.ic_linear_layout)
+    }
+
+    /**
+     * Initializes the [Menu] to be used with the current [Activity]
+     *
+     * The signature of this function has also changed slightly when changing from
+     * an Activity to a Fragment
+     */
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.layout_menu, menu)
+
+        val layoutButton = menu.findItem(R.id.action_switch_layout)
+        // Calls code to set the icon based on the LinearLayoutManager of the RecyclerView
+        setIcon(layoutButton)
+    }
+
+    /**
+     * Determines how to handle interactions with the selected [MenuItem]
+     */
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_switch_layout -> {
+                // Sets isLinearLayoutManager (a Boolean) to the opposite value
+                isLinearLayoutManager = !isLinearLayoutManager
+                // Sets layout and icon
+                chooseLayout()
+                setIcon(item)
+
+                return true
+            }
+            //  Otherwise, do nothing and use the core event handling
```

Comment:
please format

Reply:
Can you clarify? Merge with the following comment block? I didn't write the original code so not sure of the intent.

Reply:
Got it. Looks like we should be using one space for indent.